isdproject/src/ShoppingCart.js

```diff
@@ -1,58 +1,220 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import Cookies from 'js-cookie';
+import axios from 'axios'; // Import axios for API calls
+import { fetchProductById } from './api';
 import { Link } from 'react-router-dom';
 import './ShoppingCartPage.css';
 
 const ShoppingCartPage = () => {
-  const [cartItems, setCartItems] = useState([
-    { id: 1, name: 'Product 1', price: 10, quantity: 1 },
-    { id: 2, name: 'Product 2', price: 15, quantity: 2 },
-    { id: 3, name: 'Product 3', price: 20, quantity: 1 },
-  ]);
+  const [cartItems, setCartItems] = useState([]);
+  useEffect(() => {
+    const fetchCartItems = async () => {
+      try {
+        const cartString = Cookies.get('cart') || '{}';
+        const cart = JSON.parse(cartString);
+        const productIds = Object.keys(cart);
+        const itemsFromCart = [];
+
+        for (const productId of productIds) {
+          const product = await fetchProductById(productId);
+          itemsFromCart.push({ ...product, quantity: cart[productId] });
+        }
+
+        setCartItems(itemsFromCart);
+      } catch (error) {
+        console.error('Error fetching cart items:', error);
+      }
+    };
+
+    fetchCartItems();
+  }, []);
+
+  useEffect(() => {
+    const fetchSampleCartItems = () => {
+      // Sample data for cart items
+      const sampleCartItems = [
+        { id: 1, name: ' Product 1', price: 10, quantity: 1 },
+        { id: 2, name: ' Product 2', price: 15, quantity: 1 },
+        { id: 3, name: ' Product 3', price: 20, quantity: 1 },
+        // Add more sample items as needed
+      ];
+
+      if (cartItems.length === 0) {
+        setCartItems(sampleCartItems);
+      }
+    };
+
+    fetchSampleCartItems();
+  }, [cartItems]);
+
+  useEffect(() => {
```

Comment:
This page should just be for building the cart where Shipping and Payment Object haven't been created yet. So instead of creating the objects on each new cart update, we will be running createShipment() and createPayment() on Checkout processing.